components/datetime/src/options/components.rs

```diff
@@ -86,6 +86,7 @@ use serde::{Deserialize, Serialize};
 /// See the [module-level](./index.html) docs for more information.
 #[derive(Debug, Clone, PartialEq, Default, Copy)]
 #[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+#[allow(clippy::exhaustive_structs)] // this type is stable
```

Comment:
Nit: I would make this `non_exhaustive` because I think this struct is rarely going to be exhaustively constructed, and it's quite easy to see situations where we may wish to change it.